### ISSUE TITLE
ImagePicker.openCropper make the app crash on Android, BitmapWorkerTask: Invalid Uri scheme null

### ISSUE THREAD

**Issue by thierryfornetti** [ivpusic/react-native-image-crop-picker#600]
### Version
Tell us which versions you are using: 

- react-native-image-crop-picker v0.19.0
- react-native v0.49.5

### Platform
Tell us to which platform this issue is related

- Android

Tested on Device:
- Huawei ALE-L21 with Android 5.0.1
- Samsung SM-A310F with Android 7.0

### Expected behaviour
Starting from the base64 code of an image, I save the image on local storage with [itinance/react-native-fs](https://github.com/itinance/react-native-fs), then I want to use ImagePicker.openCropper to open the image given the path and crop it.

### Actual behaviour
App crash, it close without any message. The log retrieved with `adb logcat` is
```
I/art     (22627): Can not find class: Landroid/view/SearchEvent;
W/art     (22627): Before Android 4.1, method android.graphics.PorterDuffColorFilter android.support.graphics.drawable.VectorDrawableCompat.updateTintFilter(android.graphics.PorterDuffColorFilter, android.content.res.ColorStateList, android.graphics.PorterDuff$Mode) would have incorrectly overridden the package-private method in android.graphics.drawable.Drawable
I/HwCust  (22627): Constructor found for class android.widget.HwCustTextViewImpl

E/BitmapWorkerTask(22627): Invalid Uri scheme null

E/AndroidRuntime(22627): FATAL EXCEPTION: AsyncTask #1
E/AndroidRuntime(22627): Process: com.myName.myApp, PID: 22627
E/AndroidRuntime(22627): java.lang.RuntimeException: An error occured while executing doInBackground()
E/AndroidRuntime(22627): 	at android.os.AsyncTask$3.done(AsyncTask.java:300)
E/AndroidRuntime(22627): 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
E/AndroidRuntime(22627): 	at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
E/AndroidRuntime(22627): 	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
E/AndroidRuntime(22627): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
E/AndroidRuntime(22627): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/AndroidRuntime(22627): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/AndroidRuntime(22627): 	at java.lang.Thread.run(Thread.java:831)
E/AndroidRuntime(22627): Caused by: java.lang.IllegalArgumentException: Invalid Uri schemenull
E/AndroidRuntime(22627): 	at com.yalantis.ucrop.task.BitmapLoadTask.processInputUri(BitmapLoadTask.java:187)
E/AndroidRuntime(22627): 	at com.yalantis.ucrop.task.BitmapLoadTask.doInBackground(BitmapLoadTask.java:93)
E/AndroidRuntime(22627): 	at com.yalantis.ucrop.task.BitmapLoadTask.doInBackground(BitmapLoadTask.java:44)
E/AndroidRuntime(22627): 	at android.os.AsyncTask$2.call(AsyncTask.java:288)
E/AndroidRuntime(22627): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
E/AndroidRuntime(22627): 	... 4 more
```

### Steps to reproduce

1. Start with the base64 of an image
```
asset: {
   mime: "image/jpeg",
   data: ""/9j/4AAQSkZJRgABAgEBLAEsAAD...",
}
```

2. Use [itinance/react-native-fs](https://github.com/itinance/react-native-fs) to save the image on local storage, then try to use openCropper to crop the image
```
const path = `${RNFS.CachesDirectoryPath}/${image.id}`;
// path = /storage/emulated/0/Pictures/DSC_0002.jpg

// write the file
RNFS.writeFile(path, image.asset.data, "base64")
.then((success) => {
    console.log('FILE WRITTEN! '+path, success );  

    return ImagePicker.openCropper({ // <<<<< Here  the app crash
        path: path,
        width: 1024,
        height: 1024,
        cropping: true,
        includeBase64: true,
    });
})
.then((croppedImageAsset) => {
    console.log("cropImage croppedImageAsset", croppedImageAsset);
    this.props.navigation.state.params.onContentSelected( croppedImageAsset);
    // Now delete image from cache
    return RNFS.unlink(path)
})
.then(() => {
    console.log('FILE DELETED');
})
.catch((err) => {
    console.log("cropImage Error", err.message);
});
```

### Fixes
This looks related to Yalantis/uCrop#347, I found 2 ways to solve this:
 - Apply the solution provided in the issue, in [PickerModule.java](https://github.com/ivpusic/react-native-image-crop-picker/blob/master/android/src/main/java/com/reactnative/ivpusic/imagepicker/PickerModule.java) change:
```
    public void openCropper(final ReadableMap options, final Promise promise) {
        final Activity activity = getCurrentActivity();
        ...
        Uri uri = Uri.parse(options.getString("path"));
        startCropping(activity, uri);
    }
```
into:
```
    public void openCropper(final ReadableMap options, final Promise promise) {
        final Activity activity = getCurrentActivity();
        ...
        Uri uri = Uri.fromFile(new File((options.getString("path"))));
        startCropping(activity, uri);
    }
```
 - Basically `Uri.fromFile` change the path from `/storage/emulated/0/Pictures/DSC_0002.jpg` to `file:///storage/emulated/0/Pictures/DSC_0002.jpg`, so I just added the missing string:
```
ImagePicker.openCropper({
    path: `file://${path}`,
    width: 1024,
    height: 1024,
    cropping: true,
    includeBase64: true,
})
```
I tested with my android devices and with ios emulator and it works.

**Comment by ivpusic** [comment]
could you send a PR for this?

**Comment by thierryfornetti** [comment]
I'm actually not sure if this is an error, maybe I'm sending a wrong `path`, otherwise `openCropper` would never work for anyone and I didn't found any issues about it.
Before changing `PickerModule` we should understand which is the correct way to create the uri for the method `startCropping`.

**Comment by ivpusic** [comment]
there is already an working example of using cropper. It is located in example project here https://github.com/ivpusic/react-native-image-crop-picker/blob/master/example/app.js#L88

**Comment by JongLim107** [comment]
I also came into this issue.
For me, change to  'compileSdkVersion 27' and use 'com.android.support:appcompat-v7:27.0.2' can fix this issue.

**Comment by jordangrant** [comment]
I am getting this when feeding the 
```
ImagePicker.openCropper({
        path: path,
```
an image from the res/drawable folder written like 'applogo'
not sure what filepath to feed it instead. Hmm...

**Comment by punisher97** [comment]
I fixed change 
```Java 
final Uri uri = Uri.parse(options.getString("path"));
``` 
to


```Java
    @ReactMethod
    public void openCropper(final ReadableMap options, final Promise promise) {
        final Activity activity = getCurrentActivity();

        if (activity == null) {
            promise.reject(E_ACTIVITY_DOES_NOT_EXIST, "Activity doesn't exist");
            return;
        }

        setConfiguration(options);
        resultCollector.setup(promise, false);

        final Uri uri = Uri.parse(options.getString("path"));
...
```
